dev/update-examples.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+set -e
+cd "$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"/../
+
+ROOT=`pwd`
+INDEX=$ROOT/examples/README.md
+INSERT_LINE=6
+
+rm -f $INDEX
+touch $INDEX
+
+echo "# Flower Examples documentation" >> $INDEX
+echo "" >> $INDEX
+echo "\`\`\`{toctree}" >> $INDEX
+echo "---" >> $INDEX
+echo "maxdepth: 1" >> $INDEX
+echo "---" >> $INDEX
+
+rm -f "examples/doc/source/*.md"
+
+cd examples/
+for d in $(printf '%s\n' */ | sort -V); do
+  example=${d%/}
+  [[ $example = doc ]] || cp $example/README.md $ROOT/examples/doc/source/$example.md 2>&1 >/dev/null
+  [[ $example = doc ]] || (echo $INSERT_LINE; echo a; echo $example; echo .; echo wq) | ed $INDEX 2>&1 >/dev/null
+done
```

Comment:
```suggestion
for d in $(printf '%s\n' */ | sort -V); do
  example=${d%/}
  [[ $example = doc ]] || cp $example/README.md $ROOT/examples/doc/source/$example.md 2>&1 >/dev/null
  [[ $example = doc ]] || (echo $INSERT_LINE; echo a; echo $example; echo .; echo wq) | ed $INDEX 2>&1 >/dev/null
done
```

Can you add few comments what happens here?